mm2src/coins/z_coin.rs

```diff
@@ -1153,15 +1202,44 @@ impl MarketCoinOps for ZCoin {
         ))
     }
 
+    /// Calculates the wallet balance, divided into spendable and unspendable portions.
+    /// Unspendable balance consists of notes that are locked in the wallet.
     fn my_balance(&self) -> BalanceFut<CoinBalance> {
         let coin = self.clone();
         let fut = async move {
-            let sat = coin
+            let spendabled_notes = coin
+                .get_spendable_notes()
+                .await
+                .map_err(|err| BalanceError::WalletStorageError(err.to_string()))?;
+
+            let locked_notes = coin
+                .z_fields
+                .locked_notes_db
+                .load_all_notes()
+                .await
+                .mm_err(|e| BalanceError::WalletStorageError(e.to_string()))?
+                .iter()
+                .map(|n| n.rseed.clone())
+                .collect::<HashSet<_>>();
+
+            let unspendable_sat = spendabled_notes
+                .into_iter()
+                .filter(|n| locked_notes.contains(&rseed_to_string(&n.rseed)))
+                .map(|n| u64::from(n.note_value))
+                .sum::<u64>();
```

Comment:
`Amount` type in zcash lib has sum implemented. Why not sum it then use try from instead of just from, just to be sure everything is correct.

Reply:
`From` implementation for `Amount` casts `i64` as `u64`, which is why I asked for using `try_from`